pinot-core/src/main/java/org/apache/pinot/core/operator/filter/AndFilterOperator.java

```diff
@@ -60,12 +63,21 @@ protected BlockDocIdSet getFalses() {
     List<BlockDocIdSet> blockDocIdSets = new ArrayList<>(_filterOperators.size());
     for (BaseFilterOperator filterOperator : _filterOperators) {
       if (filterOperator.isResultEmpty()) {
+        blockDocIdSets.add(EmptyDocIdSet.getInstance());
+        continue;
+      }
+      if (filterOperator.isResultMatchingAll()) {
         blockDocIdSets.add(new MatchAllDocIdSet(_numDocs));
```

Comment:
We probably don't need to add this. We can check how many `blockDocIdSets` created in the end and construct the final `DocIdSet`

Reply:
good points!